### DISCUSSION TITLE
How to set background color of code block?

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-nvim#73] [Q&A] [answered]
Using `quarto` in `vscode` to edit `qmd` file, you'll get a nice background color to highlight code blocks in the `markdown` file, as shown below, so as to differentiate them from other components in the `qmd` file.

May I ask how to config `quarto-nvim`, or `neovim`, so that we can customize the background color of code blocks enclosed in one pair of **```** symbols within the `neovim` interface itself for visual clarity (not what will be rendered out)?   

<img width="726" alt="Screenshot 2023-06-29 at 6 38 40 PM" src="https://github.com/quarto-dev/quarto-nvim/assets/23435810/a1398b05-a66e-429d-9151-774d8bff1966">

**Comment by jmbuhr** [comment, marked as answer]
Yes, you can change the highlighting to whatever highlight group you want. First, open `:Telescope highlights` to preview available highlight groups:
![image](https://github.com/quarto-dev/quarto-nvim/assets/17450586/3bb8e5a7-812a-4786-ba1a-6796a361b371)

Next, create a treesitter query file to extend available highlights and assign the highlight group you chose to fenced code blocks in markdown, or, as in the following screenshot, create an entirely new highlight group:
![image](https://github.com/quarto-dev/quarto-nvim/assets/17450586/b492b315-da24-45b2-8049-fdd27f5471c2)
When going the custom route, assign for example a background color to this highlight group:
![image](https://github.com/quarto-dev/quarto-nvim/assets/17450586/d0269b8d-15b2-4932-b0ae-b39daf423a35)

Here you go:
![image](https://github.com/quarto-dev/quarto-nvim/assets/17450586/584f48e9-36c6-4d8c-9f5b-524740b08a5c)

**Caveat**: Highlights only extend to where there is actual content. It is currently not straightforward to change the highlighting of the whole area/line as seen in your VSC screenshot.

**Reply by jmbuhr** [reply]
For a solution with an additional plugin, but with the ability to highlight full lines, check out https://github.com/lukas-reineke/headlines.nvim

**Reply by hopezh** [reply]
Thanks, I managed to install `headlines.nvim` and make it work for `.md` files but not for `.qmd` files... 

My `nvim` config is [here](https://github.com/hopezh/oatLazyVim/tree/headlines). So, it seems that my `nvim` is still unable to **_recognize_** `qmd` files, probably because I haven't set up `lsp`, `cmp`, etc properly to work with `qmd` files.  

It works for `.md` file:
![image](https://github.com/quarto-dev/quarto-nvim/assets/23435810/d5b07496-28d2-41b5-82d5-5b4b206b769e)

It doesn't work for `.qmd` file: 
![image](https://github.com/quarto-dev/quarto-nvim/assets/23435810/436262b0-d7df-4245-b5c2-6236b6be2381)

But if I change `.qmd` to `.md` it will work. 
![image](https://github.com/quarto-dev/quarto-nvim/assets/23435810/cc84e0dd-f7df-4afb-a49d-dec70016843d)

**Reply by jmbuhr** [reply]
I looked at the plugin and really liked the look, so I added it to the quarto-nvim-kickstarter config. This is how you enable it for code blocks in qmd files: https://github.com/jmbuhr/quarto-nvim-kickstarter/commit/7eb3d01d2113359820321be064ff0b0a1f378c43

**Reply by hopezh** [reply]
Thanks for the quick update. 

Following your suggestion, I finally made `headlines.nvim` work for both `.md` and `.qmd` files:

For `.md` file:
![image](https://github.com/quarto-dev/quarto-nvim/assets/23435810/7b1430ce-ebcb-4e98-b358-194996addec4)

For `quarto` `.qmd` file:
![image](https://github.com/quarto-dev/quarto-nvim/assets/23435810/e1e42110-5bb4-4adf-990f-29690eab290e)

My updated `/lua/plugins/headlines.lua` config is shown [here](https://github.com/hopezh/oatLazyVim/blob/headlines/lua/plugins/headlines.lua). 

Note that the colors for headlines and code block are defined in the [/lua/config/options.lua](https://github.com/hopezh/oatLazyVim/blob/headlines/lua/config/options.lua) file, rather than inside the `headlines.nvim` file. 

Anyway, thank you for your patience and your kind help. I'm going to enjoy `quarto` in `neovim` from now on!
